src/handlers.rs

```diff
@@ -47,14 +46,12 @@ pub fn pre_prepare(
 
     check_pre_prepare_does_not_exist(msg_log, info)?;
 
-    if state.is_primary() {
-        check_pre_prepare_matches_original_block_new(msg_log, pbft_message, info)?;
-    } else {
-        set_seq_num_and_fix_block_new_seq_num(state, msg_log, &pbft_message, info)?;
-    }
+    check_pre_prepare_matches_original_block_new(msg_log, pbft_message, info)?;
 
     set_current_working_block(state, pbft_message);
 
+    state.seq_num = info.get_seq_num();
```

Comment:
This was being done in the `set_seq_num_and_fix_block_new_seq_num` method that this PR eliminates; it is necessary because it is the only place that increments the sequence number for a secondary. The current sequence number is actually not the block number of head; it is the block number of head +1, because it's the sequence number that we are currently performing consensus on. I do agree that we should just use the block number of head + 1 as the source of truth for the sequence number though; in fact, I think we should generally use on-chain information as much as possible when the PBFT state directly mirrors it. I will make a note of this and change it in a future PR.